components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -105,7 +106,16 @@ public class NetworkMessageEncoder : IMessageEncoder {
                     };
                     notification.Encode(helperEncoder);
                     helperEncoder.Close();
-                    var notificationSize = Encoding.UTF8.GetByteCount(helperWriter.ToString());
+                    var notificationSize = 0;
+                    if (compressedPayload) {
+                        var stream = new MemoryStream(Encoding.UTF8.GetBytes(helperWriter.ToString()));
```

Comment:
Please use using or try finally to ensure dispose

Reply:
Yes, fixed